rwgame/GameWindow.cpp

```diff
@@ -31,11 +31,39 @@ void GameWindow::create(const std::string& title, size_t w, size_t h,
         throw std::runtime_error("SDL_GL_CreateContext failed: " + sdlErrorStr);
     }
 
+#include "WindowIcon.h"
+
+    Uint32 rmask, gmask, bmask, amask;
+
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+    // Big Endian
+    int offset = (windowIconBitsPerPixel == 32) ? 0 : 8;
```

Comment:
You should probably assert that `(windowIconBitsPerPixel == 24) || (windowIconBitsPerPixel == 32)` ?
(This is what the code suggests)